webapp/crashstats/crashstats/tests/test_jinja_helpers.py

```diff
@@ -186,8 +157,13 @@ def test_basic_url(self):
         assert qs["short_desc"] == ["Crash in [@ $&#;deadbeef]"]
         assert qs["op_sys"] == ["Windows"]
         assert qs["bug_type"] == ["defect"]
-        comment = f"Crash report: http://localhost:8000/report/index/{self.CRASH_ID}"
-        assert qs["comment"] == [comment]
+        comment = dedent(
```

Comment:
I switched these test assertions to build the expected value using dedent and triple-quoted Python strings. The Python compiler will parse the triple-quoted and keep all the characters from the source file including the whitespace (spaces, newlines, tabs and everything) in the value. The `\` at the end of a line tells the Python compiler to ignore the newline, so that's _not_ part of the value.

When the Python interpreter runs this code, the dedent function will take that string and de-dent it by the number of leading spaces in the first line.

Thus with this source file:

```python
        comment = dedent(
            f"""\
            Crash report: http://localhost:8000/report/index/{self.CRASH_ID}

            No stack."""
        )
```

the `comment` variable points to this string value:

```
Crash report: http://localhost:8000/report/index/70dda764-a402-4ca3-b806-c38dd0240328

No stack.
```